app/Jobs/CreateQueryserviceBatchesJob.php

```diff
@@ -77,7 +77,9 @@ private function tryToAppendEntitesToExistingBatches(array $entityIdsFromEvents,
             ['pending_since', '=', null],
             ['failed', '=', false],
             ['wiki_id', '=', $wikiId],
-        ])->get();
+        ])
+            ->lockForUpdate()
```

Comment:
If we'd be very pedantic we could now wrap this method in sth that checks whether the code is actually run in a transaction, and if nor running in a transaction, create one for us.

Then again this is a private method and the caller is creating the transaction for us, so I would probably not do it for avoid that overengineering stench.

Reply:
Or a slightly easier way would be having this method check whether we're in a transaction, and throw an Exception in case we're not. Should prevent callers from accidentally using this elswhere.

Reply:
I agree with the whole notion, the use of the method is pretty confined so I think the danger of using it outside of an transaction context is not too high, but I also like the foresight and the latter suggestion definitely sounds like a useful guardrail

Reply:
Then let's just have it throw an exception when there is no ongoing transaction.